hack/generated/pkg/armclient/raw_client.go

```diff
@@ -102,6 +102,9 @@ func (c *Client) PutDeployment(ctx context.Context, deployment *Deployment) (*De
 		autorest.ByUnmarshallingJSON(deployment),
 		autorest.ByClosing())
 	if err != nil {
+		// TODO: rethink how to do this
```

Comment:
super minor: Comment would be clearer if instead of "this" you said specifically what had to be re-thought. I only say this because I used to do this to myself all the time and then when I came back I'm always like... I don't see what the problem is, what did I want to fix?

Reply:
Presumably I assume the issue here is that the ID is set even if we've failed, which seems weird?

Reply:
A few things: We need the ID in the caller, which is why I added this, but also the ID doesn't need to be read from ARM since we know how to produce it. We could add a function to the `Deployment` type instead?

Reply:
Just using `GetEntityPath` in caller.